src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/WebSocketHandle.Managed.cs

```diff
@@ -44,115 +44,154 @@ public void Abort()
 
         public async Task ConnectAsync(Uri uri, CancellationToken cancellationToken, ClientWebSocketOptions options)
         {
-            HttpResponseMessage? response = null;
-            SocketsHttpHandler? handler = null;
-            bool disposeHandler = true;
-            try
+            SocketsHttpHandler handler = SetupHandler(options);
+            await ConnectAsyncHelper(uri, new HttpMessageInvoker(handler), cancellationToken, options).ConfigureAwait(false);
+        }
+
+        public async Task ConnectAsync(Uri uri, HttpMessageInvoker handler, CancellationToken cancellationToken, ClientWebSocketOptions options)
+        {
+            await ConnectAsyncHelper(uri, handler, cancellationToken, options).ConfigureAwait(false);
+        }
```

Comment:
We shouldn't need these helpers.  You can just have:
```C#
public async Task ConnectAsync(Uri uri, HttpMessageInvoker? handler, CancellationToken cancellationToken, ClientWebSocketOptions options)
{
    handler ??= new HttpMessageInvoker(SetupHandler(options));
    ...
}
```
Having the extra layer of async method adds cost, and in this case that cost doesn't buy us commensurate value.